sympy/polys/multivariate_resultants.py

```diff
@@ -0,0 +1,341 @@
+        product = prod(list_of_products)
+        product = Poly(product).monoms()
+
+        return monomial_deg(*product)
+
+    def get_dixon_matrix(self, polynomial):
+        r"""
+        Construct the Dixon matrix from the coefficients of polynomial \alpha.
+        Each coefficient is viewed as a polynomial of x_1,..., x_n.
+        """
+
+        # A list of coefficients (in x_i, ..., x_n terms) of the power products
+        # a_1, ..., a_n in Dixon's polynomial.
+        coefficients = polynomial.coeffs()
+
+        monomials = list(itermonomials(self.variables, self.get_upper_degree()))
+        monomials = sorted(monomials, key=monomial_key('lex', self.variables))[::-1]
+
+        dixon_matrix = Matrix([[Poly(c, *self.variables).coeff_monomial(m) for m in monomials]
+                               for c in coefficients])
+
+        keep = [column for column in range(dixon_matrix.shape[-1])
+                if any([element != 0 for element in dixon_matrix[:, column]])]
+        return dixon_matrix[:, keep]
+
+class MacaulayResultant():
+    """
+    A class for calculating the Macaulay resultant. Note that the coefficients
+    of the polynomials must be given as symbols.
+
+    Examples
+    ========
+    >>> from sympy.core import symbols
+
+    >>> from sympy.polys.multivariate_resultants import MacaulayResultant
+    >>> x, y, z = symbols('x, y, z')
+
+    >>> a_0, a_1, a_2 = symbols('a_0, a_1, a_2')
+    >>> b_0, b_1, b_2 = symbols('b_0, b_1, b_2')
+    >>> c_0, c_1, c_2,c_3, c_4 = symbols('c_0, c_1, c_2, c_3, c_4')
+
+    >>> f = a_0 * y -  a_1 * x + a_2 * z
+    >>> g = b_1 * x ** 2 + b_0 * y ** 2 - b_2 * z ** 2
+    >>> h = c_0 * y - c_1 * x ** 3 + c_2 * x ** 2 * z - c_3 * x * z ** 2 + c_4 * z ** 3
+
+    >>> mac = MacaulayResultant(polynomials=[f, g, h], variables=[x, y, z])
+    >>> mac.get_monomials_set()
+    >>> matrix = mac.get_matrix()
+    >>> submatrix = mac.get_submatrix(matrix)
+    >>> submatrix
+    Matrix([
+    [-a_1,  a_0,  a_2,    0],
+    [   0, -a_1,    0,    0],
+    [   0,    0, -a_1,    0],
+    [   0,    0,    0, -a_1]])
+
+    Reference
+    ==========
+    1. [Bruce97]_
+    2. [Stiller96]_
+
+    See Also
+    ========
+    Notebook in examples: sympy/example/notebooks.
+    """
+    def __init__(self, polynomials, variables):
+        """
+        Parameters
+        ----------
+        variables: list
+            A list of all n variables
+        polynomials : list of sympy polynomials
+            A  list of m n-degree polynomials
+        """
+        self.polynomials = polynomials
+        self.variables = variables
+        self.n = len(variables)
+
+        # A list of the d_max of each variable.
+        self.degrees = [poly_from_expr(poly, self.variables)[0].degree()
+                        for poly in self.polynomials]
+
+        self.degree_m = self.get_degree_m()
+        self.monomials_size = self.get_size()
+
+    def get_degree_m(self):
```

Comment:
Is this used other than setting `self.degree_m`?

Reply:
No the method is only used for setting the `self.degree_m`. It is implemented as method because I wanted to use the docstring to explain how `degree_m` is calculated. Also I wanted to be able to test it.

Reply:
It should probably be called _get_degree_m then, so that it is private.

Reply:
I will do that, thank you!